frontend/src/shared/Hooks/useTextAnimation.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from "react";
+
+interface IUseTextAnimationArgs {
+  message:string,
+  speed?:number,
+  FunctionToTriggerUseContext: () => void
+}
+
+interface IUseTextAnimationReturnType {
+  animatedText: string;
+}
+
+
+export default function useTextAnimation({
```

Comment:
nit: If we could add some sort of comment explaining how to use this hook that would be useful. I'm assuming we just consume `animatedText`, but what is `FunctionToTriggerUseContext`? think we could use a different name like maybe like `callbackFn`?